### ISSUE TITLE
falling to recover detailed local shapes at predicted points, Single categorie ShapeNet-55

### ISSUE THREAD

**Issue by alekovargas** [yuxumin/PoinTr#25]
Dear Xumin Yu.

First thanks for this such nice work and paper.

Firstly, I was able to replicate your results using ShapeNet-55, and the pretrained model. 

But secondly, I performed the training of the network just using one of the categories of ShapeNet-55, specifically “bottles” that totally contains 498 .npy files. and training the network using just a single GPU.

I used the default configuration from the PoinTr.yaml replacing the batch size from 96 to 24 And max epoch to 600.

However, for this bottle class I was able only to reach CLD2 Min 0.87 and F-Score max 0.52, 

After some analysis it seems that model is doing good the center points predictions, but it is falling to recover detailed local shapes at these predicted points. Because mostly of this predicted local shapes are flat surfaces. ( see next picture)

![to_git](https://user-images.githubusercontent.com/4022229/148092347-add99eac-b265-462e-bd38-aa892801753a.png)

For me it is very important to know I can work with only one category and fewer samples as a previous step before of testing with my own data set, so do you have some previous experience training the network in a single category? it is possible that the DGCNN might not be extracting the features for a category with such a low number of samples?

Thanks in advance for your comments or any advice.
 
BR

**Comment by yuxumin** [comment]
Hi, @alekovargas 

Thanks for your interest in our work. I haven't tried to train PoinTr on *Single category* before, but it is nice to see the results from you.
From your results and descriptions, PoinTr can work well on the *coarse(center) prediction*  in your setting. However, the detail local geometry can not be correctly inferred. I think there maybe two reasons:

- In the training pipeline, local geometry is weakly supervised by an CD between the complete shapes, not the local regions. And CD loss, to some extend, is suboptimal in shape reconstruction problems. 
- The problem is too challenge. I am curious about the *drop ratio* when you evaluate the model ( and visualize the results). The model trained on fewer samples may lack of generalization ability to all incomplete patterns. And for those situations where the missing parts contain complex geometry patterns (beyond simple flat surfaces), the model may need more samples or more parameters. (the head and wings of airplane can be completed well by PoinTr, but for earphone, bottle, PoinTr performes worse)

For the first point, can we add a local shape reconstruction loss to constrain more on the local geometry? And for loss between overall shapes,  I haven't try to optimize EMD loss during the training of PoinTr, which worths a try.
For the second point, if your situation does not allow you to have many samples, i think more augmentations maybe helpful. In may codebase, i only use *random-crop* for the samples in ShapeNet-55. 

Hope it can help you.

Best!

**Comment by yuxumin** [comment]
Close it since no response. Feel free to re-open it if problems still exist

**Comment by alekovargas** [comment]
sorry for the late replay,

Yes more augmentation data solved the problem, thanks for such nice work. 

regards
